### PR TITLE
DIT-931 - Allow search by pseudo statuses

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: test/unit/uk/gov/hmrc/bindingtariffclassification/model/CaseSearchTest.scala [hunk 1]
```diff
@@ -37,7 +36,7 @@ class CaseSearchTest extends UnitSpec {
     queueId = Some("queue-id"),
     eori = Some("eori-number"),
     assigneeId = Some("assignee-id"),
-    statuses = Some(Set(NEW, OPEN)),
+    statuses = Some(Set(PseudoCaseStatus.NEW, PseudoCaseStatus.OPEN)),
```
Comment (SteveSugden) [inline test/unit/uk/gov/hmrc/bindingtariffclassification/model/CaseSearchTest.scala]: Did you deliberately not import PseudoCaseStatus to highlight in the code that these are not regular statuses?

File: test/unit/uk/gov/hmrc/bindingtariffclassification/repository/CaseRepositorySpec.scala [hunk 2]
```diff
@@ -48,15 +49,14 @@ class CaseRepositorySpec extends BaseMongoIndexSpec
 
   private val conflict = 11000
 
-  private val mongoDbProvider = new MongoDbProvider {
+  private val mongoDbProvider: MongoDbProvider = new MongoDbProvider {
```
Comment (SteveSugden) [inline test/unit/uk/gov/hmrc/bindingtariffclassification/repository/CaseRepositorySpec.scala]: type not required for private val.   (you don't have to change - just showing that I'm noticing things(

File: test/unit/uk/gov/hmrc/bindingtariffclassification/model/CaseSearchTest.scala (hunk 1 above)
Comment (ed0906) [reply test/unit/uk/gov/hmrc/bindingtariffclassification/model/CaseSearchTest.scala]: Could easily import PseudoCaseStatus I guess

File: test/unit/uk/gov/hmrc/bindingtariffclassification/repository/CaseRepositorySpec.scala (hunk 2 above)
Comment (ed0906) [reply test/unit/uk/gov/hmrc/bindingtariffclassification/repository/CaseRepositorySpec.scala]: Intellij was complaining anyway hence why I did it

File: app/uk/gov/hmrc/bindingtariffclassification/repository/SearchMapper.scala [hunk 3]
```diff
@@ -102,6 +114,41 @@ class SearchMapper {
     "$options" -> "i"
   }
 
+  private def filteringByStatus(search: Set[PseudoCaseStatus.PseudoCaseStatus]): (String, JsValue) = {
```
Comment (pllaurado) [inline app/uk/gov/hmrc/bindingtariffclassification/repository/SearchMapper.scala]: PseudoCaseStatus.PseudoCaseStatus looks weird

Comment (ed0906) [reply app/uk/gov/hmrc/bindingtariffclassification/repository/SearchMapper.scala]: If I pull it out into its own file I think that should help

Comment (ed0906) [reply app/uk/gov/hmrc/bindingtariffclassification/repository/SearchMapper.scala]: Fixed

**Pull request merged**